test/parallel/test-http2-compat-serverresponse-statusmessage.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const h2 = require('http2');
+
+// Http2ServerResponse.writeHead should accept an optional status message
+
+const server = h2.createServer();
+server.listen(0, common.mustCall(function() {
+  const port = server.address().port;
+  server.once('request', common.mustCall(function(request, response) {
+    const statusCode = 200;
+    const statusMessage = 'OK';
+    const headers = {'foo-bar': 'abc123'};
+    common.expectWarning(
```

Comment:
Hmm, but then it would also need to trap the warning emitted by `/lib/http2.js`, or do some `nextTick` deferral? Registering this assertion in the callback avoids that. (Not familiar at all with these APIs so please feel free to correct me.)